nestkernel/spatial/spatial.cpp

```diff
@@ -54,6 +54,48 @@ LayerMetadata::LayerMetadata( AbstractLayerPTR layer )
 {
 }
 
+void
+LayerMetadata::slice( size_t start, size_t stop, size_t step, NodeCollectionPTR node_collection )
+{
+  // Get positions of current layer, sliced in start-stop. Because the implementation of NodeCollections sliced
+  // with step internally keeps the "skipped" nodes, positions must include the "skipped" nodes as well, so that
+  // the node indices match the position indices.
+  TokenArray new_positions;
+  for ( size_t lid = start; lid < stop; ++lid )
+  {
+    const auto pos = layer_->get_position_vector( lid );
+    new_positions.push_back( pos );
+  }
+
+  // Create new free layer with sliced positions
+  int D = layer_->get_position_vector( start ).size();
+  assert( D == 2 or D == 3 );
+  AbstractLayer* layer_local = nullptr;
+  if ( D == 2 )
+  {
+    layer_local = new FreeLayer< 2 >();
+  }
+  else if ( D == 3 )
+  {
+    layer_local = new FreeLayer< 3 >();
+  }
+  // Wrap the layer as the usual pointer types.
+  std::shared_ptr< AbstractLayer > layer_safe( layer_local );
+  NodeCollectionMetadataPTR layer_meta( new LayerMetadata( layer_safe ) );
+
+  // Set the relationship with the NodeCollection.
+  node_collection->set_metadata( layer_meta );
+  nest::get_layer( node_collection )->set_node_collection( node_collection );
+  layer_meta->set_first_node_id( node_collection->operator[]( 0 ) );
+
+  // Inherit status from current layer, but with new positions.
+  DictionaryDatum layer_dict = new Dictionary();
+  layer_->get_status( layer_dict );
+  ( *layer_dict )[ names::positions ] = ArrayDatum( new_positions );
+  ( *layer_dict )[ names::step ] = step;
+  layer_local->set_status( layer_dict );
```

Comment:
I think maybe it would be better to do this as a helper function, instead of calling `set_status`. I don't know if I like the idea of `step` being set-able, I think you should only be able to slice with step if you actually slice the nodecollection.

Reply:
See my comment above. `step` is only there to make the counting of positions equal to the counting of nodes in the NodeCollection when checking that they are of equal length. A separate helper function with mostly duplicate code from `set_status()` for this seems unnecessary.